internal/pkg/keeper/client.go

```diff
@@ -0,0 +1,179 @@
+		return false, nil
+	}
+	return true, nil
+}
+
+func (client *coreKeeperClient) HasSubConfiguration(name string) (bool, error) {
+	keyPath := client.fullPath(name)
+	resp, err := client.keeperClient.KV().Keys(keyPath)
+	if err != nil {
+		return false, fmt.Errorf("checking configuration existence from Core Keeper failed: %v", err)
+	}
+	if len(resp.Keys) == 0 {
+		return false, nil
+	}
+	return true, nil
+}
+
+// PutConfigurationToml puts a full toml configuration into Core Keeper
+func (client *coreKeeperClient) PutConfigurationToml(configuration *toml.Tree, overwrite bool) error {
+	configurationMap := configuration.ToMap()
+	err := client.PutConfiguration(configurationMap, overwrite)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (client *coreKeeperClient) PutConfiguration(config interface{}, overwrite bool) error {
+	var err error
+	if overwrite {
+		err = client.keeperClient.KV().PutKeys(client.configBasePath, config)
+	} else {
+		kvPairs := convertMapToKVPairs("", config)
+		for _, kv := range kvPairs {
+			exists, err := client.ConfigurationValueExists(kv.Key)
+			if err != nil {
+				return err
+			}
+			if !exists {
+				// Only create the key if not exists in core keeper
+				if err = client.PutConfigurationValue(kv.Key, []byte(kv.Value)); err != nil {
+					return err
+				}
+			}
+		}
+	}
+	if err != nil {
+		return fmt.Errorf("error occurred while creating/updating configuration, error: %v", err)
+	}
+	return nil
+}
+
+func (client *coreKeeperClient) GetConfiguration(configStruct interface{}) (interface{}, error) {
+	exists, err := client.HasConfiguration()
+	if err != nil {
+		return nil, err
+	}
+
+	if !exists {
+		return nil, fmt.Errorf("the Configuration service (EdgeX Keeper) doesn't contain configuration for %s", client.configBasePath)
+	}
+
+	resp, err := client.keeperClient.KV().Get(client.configBasePath)
+	if err != nil {
+		return nil, err
+	}
+
+	err = decode(client.configBasePath+api.KeyDelimiter, resp.KVs, configStruct)
+	if err != nil {
+		return nil, err
+	}
+	return configStruct, nil
+}
+
+func (client *coreKeeperClient) WatchForChanges(updateChannel chan<- interface{}, errorChannel chan<- error, configuration interface{}, waitKey string) {
+}
+
+func (client *coreKeeperClient) StopWatching() {}
+
+func (client *coreKeeperClient) ConfigurationValueExists(name string) (bool, error) {
+	keyPath := client.fullPath(name)
+	res, err := client.keeperClient.KV().Keys(keyPath)
+	if err != nil {
+		return false, fmt.Errorf("checking configuration existence from Core Keeper failed: %v", err)
+	}
+	if len(res.Keys) == 0 {
+		return false, nil
+	}
+	return true, nil
+}
+
+func (client *coreKeeperClient) GetConfigurationValue(name string) ([]byte, error) {
+	keyPath := client.fullPath(name)
+	resp, err := client.keeperClient.KV().Get(keyPath)
+	if err != nil {
+		return nil, err
+	}
+	if len(resp.KVs) == 0 {
+		return nil, fmt.Errorf("%s configuration not found", name)
+	}
+
+	value := resp.KVs[0].Value.(string)
```

Comment:
always add error check for the type coversion

Reply:
Got it, done.